### PR TITLE
[dcmtk] remove ICU

### PR DESCRIPTION
**Pull request by bowenxuuu**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] <s>SHA512s are updated for each updated download.</s>
- [ ] <s>The "supports" clause reflects platforms that may be fixed by this new version.</s>
- [ ] <s>Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.</s>
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/dcmtk/dependencies.diff [hunk 1]
```diff
@@ -1,55 +1,54 @@
...
+   if(DCMTK_WITH_ICONV)
+     find_package(Iconv QUIET)
+-    find_package(LIBCHARSET QUIET)
++    set(ICONV_FOUND 1)
++    if(CHARSET_LIBRARIES)
++      set(LIBCHARSET_FOUND 1)
++    endif()
+     if(ICONV_FOUND)
+         if(NOT Iconv_IS_BUILT_IN)
+             set(LIBICONV_FOUND ${ICONV_FOUND})
```
Comment (MonicaLiu0311) [inline ports/dcmtk/dependencies.diff:54]: Looks like the EOL has changed?

Comment (xb284524239) [reply ports/dcmtk/dependencies.diff:54]: > Looks like the EOL has changed?

PS> git diff --ignore-space-at-eol | out-file -enc ascii C:/dependencies.diff

This is the command which I used.

**Comment by MonicaLiu0311** [comment]
May I ask why do we need to remove ICU?

**Comment by xb284524239** [comment]
> May I ask why do we need to remove ICU?

@MonicaLiu0311 

Because in version 3.6.9, the DCMTK official removed the support for ICU. 
In this port, there are some configurations related to ICU. I will delete them to avoid misleading users.

https://forum.dcmtk.org/viewtopic.php?t=5429

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**